fecgov/fec-cms
# Feature/4170 remove unused css

## Summary

- Resolves #4170 

Our homepage was loading elections.css, which is a huge file.
Inspector's coverage report says we're using ~13% of it.
So let's get rid of 85% of it!

Here's a possible approach—whatcha think?

### Required reviewers

Two

## Impacted areas of the application

The homepage's styles / appearance and our build process.

- New file `fec/static/scss/home.scss` is a duplicate of elections.css.
- New file `fec/home/templates/sample-homepage-for-purgecss.html` is a copy and paste of a compiled homepage with notes / instructions at the top for how this all works.
- New Gulp task ('purgecss')
- New npm script `build-homepage-css` builds it.

## Screenshots

Inspector's coverage should no longer list elections-*css. It should include home-*.css, but the Unused Bytes should be way less than 87%
![image](https://user-images.githubusercontent.com/26720877/117356339-99d7c100-ae81-11eb-90b6-e035f5d82dc1.png)

Here's the 'before':
![image](https://user-images.githubusercontent.com/26720877/117356419-b2e07200-ae81-11eb-83bc-f78a9d7446f9.png)


No visual changes:
![image](https://user-images.githubusercontent.com/26720877/117354059-e968bd80-ae7e-11eb-9140-2caa2383e0cd.png)

## Related PRs

None

## How to test

- pull the branch
- `npm i` to install the new package (gulp-purgecss)
- `npm run build` to build the css files, including the new home.css
- `npm run build-homepage-css` to take home.css and throw away everything not needed, according to the new sample homepage template
- `./manage.py runserver`
- Check every detail of the homepage. Don't forget:
  - Multiple widths
  - The skip navigation link (Should be highlighted on first tab, but you can tab backward from the top nav, too)
  - Blue dev banner at the top
  - US Gov banner
  - Top nav, open
  - Top nav, closed
  - Red announcement banners under the hero
  - Search box
  - Typeaheads
  - Rollovers
  - Interactive components
  - Feedback
  - Glossary
  - Bottom nav

Lighthouse should no longer tell us to remove unused css (because of elections-*.css _or_ home-*.css)
